tests/reftest/mod.rs

```diff
@@ -96,8 +96,7 @@ fn reftest_decoder<T: std::io::Read>(mut decoder: jpeg::Decoder<T>, path: &Path,
         let diff = (a as isize - b as isize).abs();
         max_diff = cmp::max(diff, max_diff);
 
-        // FIXME: Only a diff of 1 should be allowed?
-        if (info.coding_process != jpeg::CodingProcess::Lossless && diff <= 2) || diff == 0  {
+        if (info.coding_process != jpeg::CodingProcess::Lossless && diff <= 3) || diff == 0  {
```

Comment:
The SIMD implementation should be pretty much equivalent to the non-SIMD implementation, except for the number of fractional bits used (4 instead of 12, AFAIU) to fit in a 16-bit integer.

How did you produce the "reference" PNGs? Given how under-defined JPEG decoding is, I wouldn't be surprised at all if libjpeg-turbo produces results that are off from the "ideal" values in slightly different ways...

Reply:
Via `convert`, so image magick. Except I'm not 100% sure about all of the setings used in those calls.

Reply:
By default imagemagick uses the system libjpeg, usually libjpeg-turbo, which by default uses SIMD - so this likely ended up running the avx2 algorithm of libjpeg-turbo, which is a (different...) 8x8 16-bit IDCT. I'm not actually sure how to convince libjpeg-turbo to use the floating point IDCT...

Either way, IIRC the "official" definition of conformance is "if you DCT the pixels back, before applying any colour transforms, you stay within the quantization buckets of the encoded coefficients", so a bound of 3 here is likely fine :D